web/src/pages/PreviewPage/PreviewPage.jsx

```diff
@@ -0,0 +1,97 @@
+import React, { useRef, useState, useEffect } from 'react';
+import { useLocation, useNavigate } from 'react-router-dom';
+import { useReactToPrint } from 'react-to-print';
+import styles from './PreviewPage.module.css';
+import Header from '../../components/Header/Header.jsx';
+import CVPreview from '../../components/CVPreview/CVPreview.jsx';
+
+const PreviewPage = () => {
+  const { state } = useLocation();
+  console.log(state)
+  const navigate = useNavigate();
+  const printRef = useRef();
+
+  const [isPrinting, setIsPrinting] = useState(false);
+  const { cvData, personalInfo } = state || {};
+
+  const [currentCvData, setCurrentCvData] = useState(() => cvData || null);
+  const [currentPersonalInfo, setCurrentPersonalInfo] = useState(() => personalInfo || {});
+  console.log(typeof currentCvData)
+  console.log(personalInfo)
+  const promiseResolveRef = useRef(null);
+
+
+  if (!currentCvData) {
+  return (
+    <div style={{ padding: '2rem', color: 'red', textAlign: 'center' }}>
+      <h2>Something went wrong.</h2>
+      <p>We couldn’t load your enhanced CV. Please try again.</p>
+    </div>
+  );
+}
+  const transformedCvData = {
+  fullName: currentPersonalInfo?.fullName || '',
+  contact: {
+    email: currentPersonalInfo?.email || '',
+    phone: currentPersonalInfo?.phone || '',
+    linkedin: currentPersonalInfo?.linkedin || '',
+    github: currentPersonalInfo?.github || '',
+    portfolio: currentPersonalInfo?.portfolio || ''
+  },
+  professional_summary: currentCvData?.professionalSummary || '',
+  skills: currentCvData?.skills || [],
+  experience: currentCvData?.experience || [],
+  projects: currentCvData?.projects || [],
+  education: currentCvData?.education || [],
+  yourProfile_vs_jobCriteria: currentCvData?.yourProfile_vs_jobCriteria || ''
+};
+
+
+  useEffect(() => {
+    if (isPrinting && promiseResolveRef.current) {
+      promiseResolveRef.current();
+    }
+  }, [isPrinting]);
+
+  const handleCvSave = (updatedCvData) => {
+    setCurrentCvData(updatedCvData);
+  };
+
+  const handlePrint = useReactToPrint({
+    contentRef: printRef,
+    documentTitle: `${currentCvData?.fullName || 'CV'}_CV`,
+    onBeforePrint: () => {
+      return new Promise((resolve) => {
+        promiseResolveRef.current = resolve;
+        setIsPrinting(true);
+      });
+    },
+    onAfterPrint: () => {
+      promiseResolveRef.current = null;
+      setIsPrinting(false);
+    }
+  });
+
+  if (!currentCvData) return null;
```

Comment:
if there is no `currentCvData` we should show a better message than just `null` which might lead to confusion in the future